### ISSUE TITLE
Causal version model

### ISSUE THREAD

**Issue by hsiaohan0827** [JusperLee/TDANet#6]
Hi, thanks for releasing the code scripts. It's really useful for me.
I noticed that you have implemented TDANet of causal version in the paper, as shown in Appendix D.
I'm trying to reproduce one, but not sure about the causal implementation of avg_pooling and interpolation.
It would be really helpful if you can share your causal version TDANet model script.
Thank you very much!

**Comment by JusperLee** [comment]
Yes, I've implemented a causal version of the TDANet model in accordance with the paper, but the causal version cannot be open-sourced due to patents.

For Pooling and interpolate operations, we use torch.nn.AvgPool1d and torch.nn.functional.interpolate() to implement and add padding to prevent seeing future information.

You can refer to the implementation in this file: [causal_improved_sudormrf_v3.py](https://github.com/etzinis/sudo_rm_rf/blob/master/sudo_rm_rf/dnn/models/causal_improved_sudormrf_v3.py).
